File: Prova/Medalhas.c
#include <stdio.h>

#define MAX_PAISES 1000

typedef struct {
    char nome[101];
    int ouro;
    int prata;
    int bronze;
} Medalha;

int comparaMaior(Medalha a, Medalha b) {
    if (a.ouro < b.ouro) return 1;
    if (a.ouro > b.ouro) return 0;
    if (a.prata < b.prata) return 1;
    if (a.prata > b.prata) return 0;
    if (a.bronze < b.bronze) return 1;
    if (a.bronze > b.bronze) return 0;
    int i = 0;
    while (a.nome[i] != '\0' && b.nome[i] != '\0') {
        if (a.nome[i] < b.nome[i]) return 0;
        if (a.nome[i] > b.nome[i]) return 1;
        i++;
    }
    return a.nome[i] != '\0';
}

int main() {
    int N;
    scanf("%d", &N);
    Medalha time[MAX_PAISES];

    for (int i = 0; i < N; i++) {
        scanf("%s %d %d %d", time[i].nome, &time[i].ouro, &time[i].prata, &time[i].bronze);
    }

    for (int i = 0; i < N - 1; i++) {
        for (int j = 0; j < N - i - 1; j++) {
            if (comparaMaior(time[j], time[j + 1])) {
                Medalha temp = time[j];
                time[j] = time[j + 1];
                time[j + 1] = temp;
            }
        }
    }

    for (int i = 0; i < N; i++) {
        printf("%s %d %d %d\n", time[i].nome, time[i].ouro, time[i].prata, time[i].bronze);
    }

    return 0;
}

File: Prova/Medalhas.java
import java.util.Scanner;

class Medalha {
    String nome;
    int ouro;
    int prata;
    int bronze;

    public Medalha(String nome, int ouro, int prata, int bronze) {
        this.nome = nome;
        this.ouro = ouro;
        this.prata = prata;
        this.bronze = bronze;
    }
}

public class Medalhas {

    // retorna true se A deve vir depois de B no ranking (ou seja, precisa trocar)
    public static boolean comparaMaior(Medalha a, Medalha b) {
        if (a.ouro < b.ouro) return true;   // quem tem menos ouro vai pra trás
        if (a.ouro > b.ouro) return false;

        if (a.prata < b.prata) return true;
        if (a.prata > b.prata) return false;

        if (a.bronze < b.bronze) return true;
        if (a.bronze > b.bronze) return false;

        // empate nas medalhas -> ordem alfabética crescente
        int min = (a.nome.length() < b.nome.length() ? a.nome.length() : b.nome.length());

        for (int i = 0; i < min; i++) {
            if (a.nome.charAt(i) < b.nome.charAt(i)) return false;
            if (a.nome.charAt(i) > b.nome.charAt(i)) return true;
        }
        return a.nome.length() > b.nome.length();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt(); // lê a quantidade de países
        Medalha[] time = new Medalha[N];

        for (int i = 0; i < N; i++) {
            String nome = sc.next();
            int ouro = sc.nextInt();
            int prata = sc.nextInt();
            int bronze = sc.nextInt();
            time[i] = new Medalha(nome, ouro, prata, bronze);
        }

        // bubble sort
        for (int i = 0; i < N - 1; i++) {
            for (int j = 0; j < N - i - 1; j++) {
                if (comparaMaior(time[j], time[j + 1])) {
                    Medalha temp = time[j];
                    time[j] = time[j + 1];
                    time[j + 1] = temp;
                }
            }
        }

        for (int i = 0; i < N; i++) {
            System.out.println(time[i].nome + " " + time[i].ouro + " " +
                               time[i].prata + " " + time[i].bronze);
        }

        sc.close();
    }
}
